model/position.go

```diff
@@ -0,0 +1,185 @@
+package model
+
+import (
+	"time"
+)
+
+type MarginMode string
+
+const (
+	Isolated MarginMode = "ISOLATED"
+	Cross    MarginMode = "CROSS"
+
+	MAKER_FEE = 0.02
+	TAKER_FEE = 0.04
+)
+
+type Position struct {
+	ID            int64    `db:"id" json:"id" gorm:"primaryKey,autoIncrement"`
+	ExchangeID    int64    `db:"exchange_id" json:"exchange_id"`
+	Pair          string   `db:"pair" json:"pair"`
+	Side          SideType `db:"side" json:"side"`
+	EntryPrice    float64  `db:"entry_price" json:"entry_price"`
+	AvgPrice      float64  `db:"avg_price" json:"avg_price"`
+	Quantity      float64  `db:"quantity" json:"quantity"` //Asset e.g BTC
+	Size          float64  `db:"size" json:"size"`         // Quote e.g USDT
+	Active        bool     `db:"active" json:"active"`
+	Safe          bool     `db:"safe" json:"safe"`
+	Leverage      int
+	CurrentCandle *Candle
+	MarginMode    MarginMode `db:"margin_mode" json:"margin_mode"`
+
+	CreatedAt time.Time `db:"created_at" json:"created_at"`
+	UpdatedAt time.Time `db:"updated_at" json:"updated_at"`
+	ClosedAt  time.Time `db:"closed_at" json:"closed_at"`
+}
+
+func (p *Position) Closed() bool {
+	return p.ClosedAt.IsZero()
+}
+
+func NewPosition(pair string, side SideType, size float64, leverage int, candle Candle) Position {
+	position := Position{
+		Pair:       pair,
+		Size:       size,
+		Quantity:   size * float64(leverage),
+		Leverage:   leverage,
+		Side:       side,
+		CreatedAt:  candle.Time,
+		EntryPrice: candle.Close,
+		Safe:       false,
+	}
+
+	return position
+}
+
+type Result struct {
+	Pair          string
+	ProfitPercent float64
+	ProfitValue   float64
+	Side          SideType
+	Duration      time.Duration
+	CreatedAt     time.Time
+}
+
+func (p *Position) OppositeSide() SideType {
+	if p.Side == SideTypeBuy {
+		return SideTypeSell
+	}
+
+	return SideTypeBuy
+}
+
+func (p *Position) Update(order *Order) (result *Result, finished bool) {
```

Comment:
Why this function is necessary?

Reply:
I think we need to have an update position logic to controllers. The controller should monitor the positions and update them accordingly in case of market changes. For example, if the position is liquidated, the controller is responsible to update the position and set it as liquidated.

Not 100% percent sure about this.. I tried to copy the logic that you already have for the order updates.